File: clients/typescript/examples/collections-find-and-rerank-no-vectorize.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

// Get an existing collection
const client = new DataAPIClient();
const database = client.db("**API_ENDPOINT**", {
  token: "**APPLICATION_TOKEN**",
});
const collection = database.collection("**COLLECTION_NAME**");

(async function () {
  const cursor = collection.findAndRerank(
    {},
    {
      sort: {
        $hybrid: {
          $vector: [0.08, -0.62, 0.39],
          $lexical: "house hill grassy",
        },
      },
      rerankQuery: "A tree in the woods",
      rerankOn: "$lexical",
    },
  );

  // Iterate over the found documents
  for await (const result of cursor) {
    console.log(result.document);
  }
})();

File: clients/go/examples/collections-find-and-rerank-no-vectorize.go
package main

import (
	"context"
	"fmt"
	"log"

	"github.com/datastax/astra-db-go/v2/astra"
	"github.com/datastax/astra-db-go/v2/astra/cursors"
	"github.com/datastax/astra-db-go/v2/astra/filter"
	"github.com/datastax/astra-db-go/v2/astra/options"
	"github.com/datastax/astra-db-go/v2/astra/sort"
)

func main() {
	ctx := context.Background()

	// Get an existing collection
	client := astra.NewClient()

	database := client.Database(
		"**API_ENDPOINT**",
		options.API().SetToken("**APPLICATION_TOKEN**"),
	)

	collection := database.Collection("**COLLECTION_NAME**")

	// Find documents
	vectorQuery := []float32{0.08, -0.62, 0.39}
	lexicalQuery := "house hill grassy"

	cursor := collection.FindAndRerank(
		filter.F{},
		options.CollectionFindAndRerank().
			SetSort(sort.HybridBy(sort.HybridSort{
				Vector:  &vectorQuery,
				Lexical: &lexicalQuery,
			})).
			SetRerankQuery("A tree in the woods").
			SetRerankOn("$lexical"),
	)

	// Iterate over the found documents
	for cursor.Next(ctx) {
		var result cursors.RerankedResult[astra.Document]
		if err := cursor.Decode(&result); err != nil {
			log.Fatal(err)
		}
		fmt.Println(result.Document.ToMap())
	}
}
